### ISSUE TITLE
Multiple style tags with linaria

### ISSUE THREAD

**Issue by ShanonJackson** [zeit/next.js#10220]
# Bug report
Since testing upgrade to Next 9.2.0 my lighthouse performance score has lost -30 points, leading to an inability to upgrade the version from 9.1.7.

## Describe the bug
On quick investigation this is due to next-css (the plugin i use in combination with Linaria css-in-js). Working in sync with the new inbuilt css changes and adding 20+ `<link rel="preload"...>` and `<link rel="stylesheet" .../>` tags into the headers; Iv'e attached screenshots at the bottom.

This leads to the lighthouse warning: "Peyload key requests" Potential sayings "0.6s"
"A preload <link> was found for "https://MYWEBSITEHERE/_next/static/css/bdc370faa38e17f08cb420bf2435de212a3f524f.81ef22a8.chunk.css" but was not used by the browser. Check that you are using the `crossorigin` attribute properly."

## To Reproduce
Haven't tested but can imagine its easily reproducible in a couple of small steps.
1. Go to with-linaria example. (However this will be happening for all .css users)
2. Create a bunch of pages with styled components created in them from linaria.
3. Link to them from homepage.
4. Deploy application on now with NextJS 9.2.0+

## Expected behavior
The expected behavior is that next-css should be unaffected by the inbuilt css changes. However the desired behavior should be linaria operates with the same special permissions as css-modules in terms of allowing css imports in any file (not just _app.tsx). 

## Screenshots
https://imgur.com/a/00vJfag

## System information
Irrelevant.

## Additional context
https://github.com/zeit/next.js/issues/9931

**Comment by joonhocho** [comment]
Is it related to this issue?: https://github.com/zeit/next.js/issues/10268
Browser needs to rerender because css is not included in SSR may cause browser to flash and delay initial loading speed.

**Comment by Timer** [comment]
I could not reproduce with your instructions:
https://github.com/Timer/next-repros/tree/issue/10220

Please provide a full demo.

**Comment by ShanonJackson** [comment]
@Timer spent some time today trying to reproduce as its still happening its the difference between 9.1.7 (1-5 .css tags) and 9.2.0 like 30 link preload tags.

So the reproduction is going to be outside of the scope here because to reproduce it you actually need an external npm package where they've left in the require(".css") imports; I use my own private personal component library.

I leave the require(".css") imports explicitly so that when i import components i also automatically import that components critical css and nothing more.

Sorry i thought the problem was linaria/next at first but it seems more complicated than that.

TL;DR
The culprit is a combination of leaving in require(".css") imports in a node_modules library.
The culprit is also that running through next-transpile-modules (basically used just to enable css importing in node_modules).
The culprit is also Next 9.2.0 css changes as even running this combination on 9.1.7 everything is fine.

I don't who's responsibility it should be to fix/look into this.

@Timer Let me know if you have any questions because i would still love to have this fixed; May be possible to setup a repro but would take more time than i have available right now as i'd have to go track down some librarys that leave in the css import statements which are few and far between.

**Comment by ShanonJackson** [comment]
@Timer I'm setting up the reproduction now; Iv'e got it working on your demo app but just want to figure out why it actually happens because i had to change it significantly to look like my setup.

Once iv'e figured out why it actually happens i'll report back here later today.

**Comment by ShanonJackson** [comment]
@Timer 
https://github.com/ShanonJackson/nextcssbug

Less than standard setup so be careful; Iv'e spent most of today trying to get a good reproduction that deployed only has about 6 css chunks compared to my production app (35-40+ haven't checked exact numbers).

To save you some time flack-ui is my ui library where iv'e left in css imports require(".css") in the components themselves; Then as i import each component it will automatically import the critical css.
To achieve this i use next-transpile-modules.

I cannot replicate this bug WITHOUT next-transpile-modules; However i also cannot replicate this bug with next-transpile-modules and NextJS 9.1.7;

(Page1 -> Page4) Cause this bug because i got lazy but only with those 4 files importing from flack-ui i already get a large amount more .css statements in the head compared to NextJS 9.1.7.

TL;DR
next-transpile-modules plays a part.
doesn't happen on 9.1.7
happens on 9.2.0/9.2.1
each named import from flack-ui has a named export which imports css (require(".css")).
To make it work the named exports in each page has to change if they all share the same named exports the bug will not happen.

Let me know if you have any issues.

**Comment by 0xsven** [comment]
try turning off nextjs' internal way of dealing with CSS that was introduced in [9.2](https://nextjs.org/blog/next-9-2)

```js
// next.config.js
module.exports = withCSS({
  experimental: {
    css: false,
  },

```

**Comment by Timer** [comment]
This should be fixed in the latest `next@canary`.

**Comment by ShanonJackson** [comment]
Thanks @Timer your a champ.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
